app/services/transaction_service/gateway/braintree_adapter.rb

```diff
@@ -20,12 +22,41 @@ def create_payment(tx:, gateway_fields:, prefer_async: nil)
 
       result = BraintreeSaleService.new(payment, gateway_fields).pay(false)
 
-      unless result.success?
-        return SyncCompletion.new(Result::Error.new(result.message))
+      if !result.success?
+        SyncCompletion.new(Result::Error.new(result.message))
+      end
+
+      SyncCompletion.new(Result::Success.new({result: true}))
+    end
+
+    def reject_payment(tx:, reason: nil)
+      result = BraintreeService::Payments::Command.void_transaction(tx[:id], tx[:community_id])
+
+      if !result.success?
+        SyncCompletion.new(Result::Error.new(result.message))
       end
 
       SyncCompletion.new(Result::Success.new({result: true}))
     end
 
+    def complete_preauthorization(tx:)
+      result = BraintreeService::Payments::Command.submit_to_settlement(tx[:id], tx[:community_id])
+
+      if !result.success?
+        SyncCompletion.new(Result::Error.new(result.message))
+      end
+
+      SyncCompletion.new(Result::Success.new({result: true}))
+    end
+
+    def get_payment_details(tx:)
+      payment_total = Maybe(PaymentModel.where(transaction_id: tx[:id]).first).total_sum.or_else(nil)
+      total_price = tx[:unit_price] * 1 # TODO fixme for booking (model.listing_quantity)
```

Comment:
I need to check this once more. It's a copy & paste comment so no behavior change here between master and this branch but I could either remove the comment or then change something.